tests/contrib/openstack/test_ip.py

```diff
@@ -46,6 +47,18 @@ def _patch(self, method):
         self.addCleanup(_m.stop)
         return mock
 
+    def test_get_invalid_ips_valid_ip(self):
+        self.is_clustered.return_value = True
+        self.test_config.set('vip', '10.5.3.200')
+        self.get_iface_for_address.return_value = 'ens3'
+        self.assertEquals(ip.get_invalid_vips(), [])
+
+    def test_get_invalid_ips_invalid_ip(self):
+        self.is_clustered.return_value = True
+        self.test_config.set('vip', '10.3.2.50')
+        self.get_iface_for_address.return_value = None
+        self.assertEquals(ip.get_invalid_vips(), ['10.3.2.50'])
```

Comment:
Can you add a unit tests for:

1. vip being unset
2. Having a mix of valid and invalid vips